src/engine/Xpattern_match_regexp.ml

```diff
@@ -19,8 +19,48 @@ module MV = Metavariable
 
 let tags = Logs_.create_tags [ __MODULE__ ]
 
-let regexp_matcher ?(base_offset = 0) big_str (file : Fpath.t) regexp =
-  let subs = Pcre_.exec_all_noerr ~rex:regexp big_str in
+type ('regex, 'libregex, 'substring) re_functions = {
+  get_pattern : 'regex -> string;
+  get_regex : 'regex -> 'libregex;
+  exec_all_noerr : 'regex -> string -> 'substring array;
+  get_substring : 'substring -> int -> string;
+  get_substring_ofs : 'substring -> int -> int * int;
+  names : 'libregex -> string array;
+  num_of_subs : 'substring -> int;
+  get_named_substring_ofs : 'libregex -> string -> 'substring -> int * int;
+  get_named_substring : 'libregex -> string -> 'substring -> string;
+}
+
+let pcre2_regex_functions =
```

Comment:
can you add a comment why we need both pcre and pcre2 here? I thought the whole engine was switching to
pcre2 except spacegrep/aliengrep, so why we still need pcre here?

Reply:
Yes, it is because spacegrep/aliengrep use this. I can add a note.

Reply:
only aliengrep but yes.